near-sdk-abi/src/lib.rs

```diff
@@ -0,0 +1,113 @@
+use __private::{generate_ext, read_abi};
+use anyhow::{anyhow, Result};
+use convert_case::{Case, Casing};
+use quote::{format_ident, quote};
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::{env, fs};
+
+// Private functions shared between macro & generation APIs, not stable to be used.
+#[doc(hidden)]
+#[path = "private/mod.rs"]
+pub mod __private;
+
+pub struct AbiFile {
+    /// Path to the ABI JSON file.
+    pub path: PathBuf,
+    /// Contract name to be used for the resulting trait name.
+    /// If missing will try to pull the name from ABI metadata and use `Ext<ContractName>`.
+    pub contract_name: Option<String>,
+    /// mod name to be used for the resulting ext mod.
+    /// If missing will be derived by applying snake case to the contract name, e.g. ext_status_message.
+    pub mod_name: Option<String>,
+}
+
+impl AbiFile {
```

Comment:
should we add builder patterns to be able to chain the initialization? Intuition is that overriding the contract/module name might be a bit awkward

Reply:
It seemed a little awkward to me to have 4 different combinations of (opt_contract_name, opt_mod_name) as opposed to just 2 like in near-abi-client-rs.

My idea was to do something like `Generator::new(...).file(AbiFIle { path: ..., contract_name: None, mod_name: Some("ext_cool_contract") }`. What would be a better alternative?

Reply:
that's fair, just indicating that this might be difficult to maintain in the future because any addition of fields or changing of types will break the API